src/ParcelRegistry.Importer.Grb/Infrastructure/Program.cs

```diff
@@ -1,9 +1,183 @@
+    using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Options;
+    using ParcelRegistry.Infrastructure;
+    using ParcelRegistry.Infrastructure.Modules;
+    using Serilog;
+    using Serilog.Debugging;
+    using Serilog.Extensions.Logging;
+
+    public sealed class Program
     {
-        private static void Main(string[] args)
+        protected Program()
+        { }
+
+        public static async Task Main(string[] args)
         {
+            AppDomain.CurrentDomain.FirstChanceException += (sender, eventArgs) =>
+                Log.Debug(
+                    eventArgs.Exception,
+                    "FirstChanceException event raised in {AppDomain}.",
+                    AppDomain.CurrentDomain.FriendlyName);
+
+            AppDomain.CurrentDomain.UnhandledException += (sender, eventArgs) =>
+                Log.Fatal((Exception) eventArgs.ExceptionObject, "Encountered a fatal exception, exiting program.");
+
+            var projectName = Assembly.GetEntryAssembly().GetName().Name;
+            Log.Information($"Starting {projectName}");
+
+            var host = new HostBuilder()
+                .ConfigureAppConfiguration((hostContext, builder) =>
+                {
+                    builder
+                        .SetBasePath(Directory.GetCurrentDirectory())
+                        .AddJsonFile("appsettings.json", optional: false, reloadOnChange: false)
+                        .AddJsonFile($"appsettings.{Environment.MachineName}.json", optional: true, reloadOnChange: false)
+                        .AddEnvironmentVariables()
+                        .AddCommandLine(args);
+                })
+                .ConfigureLogging((hostContext, builder) =>
+                {
+                    SelfLog.Enable(Console.WriteLine);
+
+                    Log.Logger = new LoggerConfiguration()
+                        .ReadFrom.Configuration(hostContext.Configuration)
+                        .Enrich.FromLogContext()
+                        .Enrich.WithMachineName()
+                        .Enrich.WithThreadId()
+                        .Enrich.WithEnvironmentUserName()
+                        .Destructure.JsonNetTypes()
+                        .CreateLogger();
+
+                    builder.ClearProviders();
+                    builder.AddSerilog(Log.Logger);
+                })
+                .ConfigureServices((hostContext, services) =>
+                {
+                    var loggerFactory = new SerilogLoggerFactory(Log.Logger);
+
+                    services
+                        .AddScoped(s => new TraceDbConnection<ImporterContext>(
+                            new SqlConnection(hostContext.Configuration.GetConnectionString("Events")),
+                            hostContext.Configuration["DataDog:ServiceName"]))
+                        .AddScoped<IImporterContext, ImporterContext>()
+                        .AddDbContextFactory<ImporterContext>((provider, options) => options
```

Comment:
We don't use the `ContextFactory`. But I would revisit that. Cause this can be a long running process. Maybe remove the `.AddScoped<IImporterContext, ...>`?